forte/data/extractor/predictor.py

```diff
@@ -0,0 +1,199 @@
+                extractor = scheme["extractor"]
+                feature = extractor.extract(pack, instance)
+                feature_collection[tag] = feature
+            
+            self.pack_pools.append(pack)
+            self.instance_pools.append(instance)
+            self.features_pools.append(feature_collection)
+
+            if len(self.features_pools) == self.batch_size:
+                yield self.convert(self.features_pools[:self.batch_size]), \
+                    self.pack_pools[:self.batch_size], \
+                    self.instance_pools[:self.batch_size]
+                
+                self.features_pools.clear()
+                self.pack_pools.clear()
+                self.instance_pools.clear()
+
+    def flush_batch(self):
+        if len(self.features_pools) > 0:
+            yield self.convert(self.features_pools), self.pack_pools, \
+                self.instance_pools
+            self.features_pools.clear()
+            self.pack_pools.clear()
+            self.instance_pools.clear()
+
+    def _process(self, input_pack: PackType):
+        for tensor_collection, packs, instances in self.yield_batch(input_pack):
+            predictions = self.predict(tensor_collection)
+            self.add_to_pack(predictions, tensor_collection, packs, instances)
+
+        # update the status of the jobs. The jobs which were removed from
+        # data_pack_pool will have status "PROCESSED" else they are "QUEUED"
+        q_index = self._process_manager.current_queue_index
+        u_index = self._process_manager.unprocessed_queue_indices[q_index]
+        data_pool_length = len(set(self.pack_pools))
+        current_queue = self._process_manager.current_queue
+
+        for i, job_i in enumerate(
+                itertools.islice(current_queue, 0, u_index + 1)):
+            if i <= u_index - data_pool_length:
+                job_i.set_status(ProcessJobStatus.PROCESSED)
+            else:
+                job_i.set_status(ProcessJobStatus.QUEUED)
+
+    def flush(self):
+        for tensor_collection, packs, instances in self.flush_batch():
+            predictions = self.predict(tensor_collection)
+            self.add_to_pack(predictions, tensor_collection, packs, instances)
+            pack = packs[0]
+
+        current_queue = self._process_manager.current_queue
+        for job in current_queue:
+            job.set_status(ProcessJobStatus.PROCESSED)
+
+    def predict(self, tensor_collection):
+        predictions = self.model(tensor_collection["text_tag"]["tensor"])
+        return predictions
+
+    def add_to_pack(self, predictions, tensor_collection, packs, instances):
+        prev_pack = None
+        for pred, mask, pack, instance in zip(predictions,
+                                        tensor_collection["text_tag"]["mask"][0],
+                                         packs, instances):
+            self.feature_scheme["ner_tag"]["extractor"].add_to_pack(
```

Comment:
I guess you won't use these hardcoded ones?

Reply:
Yes. I am discussing with Yanwen how to assess the config, so that it is in the same style in training and prediction.